packages/@aws-cdk/aws-appmesh/lib/virtual-gateway.ts

```diff
@@ -0,0 +1,230 @@
+import * as cdk from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnGatewayRoute, CfnVirtualGateway } from './appmesh.generated';
+import { GatewayRoute, GatewayRouteBaseProps } from './gateway-route';
+
+import { IMesh, Mesh } from './mesh';
+import { AccessLog } from './shared-interfaces';
+import { VirtualGatewayListener, VirtualGatewayListenerConfig } from './virtual-gateway-listener';
+
+/**
+ * Interface which all Virtual Gateway based classes must implement
+ */
+export interface IVirtualGateway extends cdk.IResource {
+  /**
+   * Name of the VirtualGateway
+   *
+   * @attribute
+   */
+  readonly virtualGatewayName: string;
+
+  /**
+   * The Amazon Resource Name (ARN) for the VirtualGateway
+   *
+   * @attribute
+   */
+  readonly virtualGatewayArn: string;
+
+  /**
+   * The mesh which the VirtualGateway belongs to
+   */
+  readonly mesh: IMesh;
+
+  /**
+   * Utility method to add a new GatewayRoute to the VirtualGateway
+   */
+  addGatewayRoute(id: string, route: GatewayRouteBaseProps): GatewayRoute;
+}
+
+/**
+ * Basic configuration properties for a VirtualGateway
+ */
+export interface VirtualGatewayBaseProps {
+  /**
+   * Name of the VirtualGateway
+   *
+   * @default - A name is automatically determined
+   */
+  readonly virtualGatewayName?: string;
+
+  /**
+   * Listeners for the VirtualGateway. Only one is supported.
+   *
+   * @default - Single HTTP listener on port 8080
+   */
+  readonly listeners?: VirtualGatewayListener[];
+
+  /**
+   * Access Logging Configuration for the VirtualGateway
+   *
+   * @default - no access logging
+   */
+  readonly accessLog?: AccessLog;
+}
+
+/**
+ * Properties used when creating a new VirtualGateway
+ */
+export interface VirtualGatewayProps extends VirtualGatewayBaseProps {
+  /**
+   * The mesh which the VirtualGateway belongs to
+   */
+  readonly mesh: IMesh;
+}
+
+abstract class VirtualGatewayBase extends cdk.Resource implements IVirtualGateway {
+  /**
+   * Name of the VirtualGateway
+   */
+  public abstract readonly virtualGatewayName: string;
+
+  /**
+   * The Amazon Resource Name (ARN) for the VirtualGateway
+   */
+  public abstract readonly virtualGatewayArn: string;
+
+  /**
+   * The name of the mesh which the VirtualGateway belongs to
+   */
+  public abstract readonly mesh: IMesh;
+
+  protected readonly listeners = new Array<VirtualGatewayListenerConfig>();
```

Comment:
Now that you've removed the `addListener(s)` method(s), I think you can safely remove this field, and make `listeners` a local variable of the `VirtualGateway` constructor.